internal/pkg/controllers_metrics/thanosreceive_metrics.go

```diff
@@ -0,0 +1,24 @@
+package controllers_metrics
```

Comment:
in terms of naming, since its internal, it could read like `internal/pkg/metrics` since all internal packages are relative to the controller anyway and we wont expect them to be used internally.

Aside from that, I'd almost keep these inside a single file since I expect the surface area to be limited anyway, but neither of these comments are blockers. Just a suggestion :)